opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -374,6 +374,16 @@ namespace Opm {
                 OPM_THROW(Opm::NumericalIssue,"Error encounted when solving well equations");
             }
 
+            auto& ebosJac = ebosSimulator_.model().linearizer().matrix();
+            if (param_.matrix_add_well_contributions_) {
+                wellModel().addWellContributions(ebosJac);
```

Comment:
Two potential problems:
 - I think we must use the auxiliary module if either of the two parameters (matrix_add_well_contributions or preconditioner_add_well_contributions) are true.
 - In the matrix adapter, the wells' apply() method is called unconditionally, but it should not be called it matrix_add_well_contributions is true, since then the contributions are already in the matrix used for the linear operator.

Reply:
You are right for the first one. I will fix that.

The second one is already handled in [StandardWell_impl.hpp#L1602](https://github.com/blattms/opm-simulators/blob/01b25f27edff9c03db0eae83811fe3501f00705a/opm/autodiff/StandardWell_impl.hpp#L1602). The matrix adapter does not know wether a well supports adding th contributions or not and therefore cannot decide whether to call apply or not.

Reply:
You are right, I did not notice that no-op path for apply().